### ISSUE TITLE
Request/Question: Assists/Wallbang in KillFeed

### ISSUE THREAD

**Issue by torriz** [rehlds/ReGameDLL_CS#336]
Would it be possible to show 1-2 in the kill feed?

1. If you killed someone through the wall (wallbang)?
2. Show assists in the killfeed (if enabled by player)?

I feel like this would be awesome to have!

And: Great job so far!
EDIT: if client.dll is needed for f.ex the 1.3 bhopping, how hard and time consuming is this reverse-engineering?

**Comment by voed** [comment]
Deathmsg has only 3 arguments: killer id, victim id and headshot, you cant add any extra info without client-side modifications

**Comment by torriz** [comment]
> Deathmsg has only 3 arguments: killer id, victim id and headshot, you cant add any extra info without client-side modifications
I've played on servers which show a player assisting in a kill. 

Any idea how THEY are doing it?

**Comment by stfkolev** [comment]
As for the assists, you can take a look at the following plugin for AMXX. https://forums.alliedmods.net/showthread.php?t=91360

**Comment by torriz** [comment]
> As for the assists, you can take a look at the following plugin for AMXX. https://forums.alliedmods.net/showthread.php?t=91360
No. That's not how. The entire goal is to have it in killfeed, it is possible. They do it on some russian servers..

**Comment by stfkolev** [comment]
I see. I misunderstood the issue itself. That's done by changing the nickname, but the killfeed itself, I'm not sure how it'll be done. Probably @voed is right:

> Deathmsg has only 3 arguments: killer id, victim id and headshot, you cant add any extra info without client-side modifications

**Comment by torriz** [comment]
You say it only have 3 arguments, but you cannot change that?

**Comment by CrazY770** [comment]
> you cant add any extra info without client-side modifications

**Comment by Nord1cWarr1or** [comment]
The killfeed is rendered by client.dll — the `DeathMsg` event only carries killer/victim/weapon/headshot, so wallbang or assist icons can't be displayed without modifying client.dll (confirmed in the thread); "assist" servers do it via AMXX plugins with workarounds. Closing as not planned — client.dll territory.

**Comment by torriz** [comment]
Okay I see the confusion, I was under the impression that this repo was for client- and serverside replacement. 

I (at the time) replaced regamedll->local client mp.dll to access the cool features offline too..
But, I get it.. If it's out of scope it's out of scope..

**Comment by Nord1cWarr1or** [comment]
@torriz For the record, since #858 the gamedll does send extra killfeed data for custom clients: the `DeathMsg` user message carries optional trailing fields after the four vanilla ones (killer/victim/headshot/weapon), controlled by `mp_deathmsg_flags` (default `"abc"` — all enabled). This includes the assisting player (`PLAYERDEATH_ASSISTANT`, flash-assists included) and kill-rarity bits — penetrated through walls (`KILLRARITY_PENETRATED`), through smoke, no-scope, killer blind, in-air, domination/revenge — see `CHalfLifeMultiplay::SendDeathMessage` in `regamedll/dlls/multiplay_gamerules.cpp`. Vanilla clients ignore the extra fields, so nothing changes for them; a custom client.dll that knows the extended format can render assist/wallbang icons in the killfeed.
